lib/templates/lab2/lab2.dart

```diff
@@ -1,19 +1,80 @@
-class Book {
-  // your code here
+// import 'package:flutter/material.dart';
+
+class Book with JsonSerializable {
+  String title;
+  String author;
+  int numberOfPages;
+
+  Book(this.title, this.author, this.numberOfPages);
+
+  void bookDetails() {
+    print("$title by $author has $numberOfPages pages");
+  }
+
 }
 
-class FictionBook {
-  // your code here
+class FictionBook extends Book{
+  String genre;
+
+  FictionBook(String title, String author, int numberOfPages, this.genre) : super(title, author, numberOfPages);
+
+  @override
+  void bookDetails() {
+    print("$genre book $title by $author has $numberOfPages pages");
+  }
 }
 
 mixin JsonSerializable {
-  // your code here
+  Map<String, dynamic> toJson() {
+    String? title;
+    String? author;
+    int? numberOfPages;
+    return {
+      'title': title,
+      'author': author,
+      'numberOfPages': numberOfPages
+    };
+  }
 }
 
-double calculateArea(){
-  // your code here
+double calculateArea({double? length, double? width, String type = 'rectangle'}) {
+  if (type == 'rectangle') {
+    return length! * width!;
+  } else if (type == 'circle') {
+    return 3.14 * length! * length!;
+  }
+  return 0;
 }
 
 class User{
-  // your code here
+  String name = '';
+  int? age;
+  String? email;
+
+  User({required String name, String? email, int? age}) {
+    this.name = name;
+    this.email = email;
+    this.age = age;
+  } 
+
+  void printUserdetails() {
+    print("Name: [$name]");
+    dynamic age = this.age ?? "N/A";
+    print("Age: [$age]");
+    String email = this.email ?? "N/A";
+    print("Email: [$email]");
+  } 
+}
+
+void main() {
+  var book = Book("Harry Potter", "J.K. Rowling", 400);
+  var fictionBook = FictionBook("Harry Potter", "J.K. Rowling", 400, "Fiction");
+  book.bookDetails();
+  fictionBook.bookDetails();
+
```

Comment:
print(book.toJson());
returned
{title: null, author: null, numberOfPages: null}

You didnt match mixin with needed class